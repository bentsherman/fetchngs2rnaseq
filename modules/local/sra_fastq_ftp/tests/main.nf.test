nextflow_process {

    name "Test process: SRA_FASTQ_FTP"
    script "../main.nf"
    process "SRA_FASTQ_FTP"

    test("Should run without failures") {

        when {
            process {
                """
                input[0] = [
                    [ id:'SRX9626017_SRR13191702', single_end:false, md5_1: '89c5be920021a035084d8aeb74f32df7', md5_2: '56271be38a80db78ef3bdfc5d9909b98' ], // meta map
                    [
                        'ftp.sra.ebi.ac.uk/vol1/fastq/SRR131/002/SRR13191702/SRR13191702_1.fastq.gz',
                        'ftp.sra.ebi.ac.uk/vol1/fastq/SRR131/002/SRR13191702/SRR13191702_2.fastq.gz'
                    ]
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
