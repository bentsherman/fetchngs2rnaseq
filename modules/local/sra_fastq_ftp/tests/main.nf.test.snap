{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "SRX9626017_SRR13191702",
                            "single_end": false,
                            "md5_1": "89c5be920021a035084d8aeb74f32df7",
                            "md5_2": "56271be38a80db78ef3bdfc5d9909b98"
                        },
                        [
                            "SRX9626017_SRR13191702_1.fastq.gz:md5,baaaea61cba4294ec696fdfea1610848",
                            "SRX9626017_SRR13191702_2.fastq.gz:md5,8e43ad99049fabb6526a4b846da01c32"
                        ]
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "SRX9626017_SRR13191702",
                            "single_end": false,
                            "md5_1": "89c5be920021a035084d8aeb74f32df7",
                            "md5_2": "56271be38a80db78ef3bdfc5d9909b98"
                        },
                        [
                            "SRX9626017_SRR13191702_1.fastq.gz.md5:md5,055a6916ec9ee478e453d50651f87997",
                            "SRX9626017_SRR13191702_2.fastq.gz.md5:md5,c30ac785f8d80ec563fabf604d8bf945"
                        ]
                    ]
                ],
                "2": [
                    "versions.yml:md5,6b60ed6d5805271a1b97798e29c0635c"
                ],
                "fastq": [
                    [
                        {
                            "id": "SRX9626017_SRR13191702",
                            "single_end": false,
                            "md5_1": "89c5be920021a035084d8aeb74f32df7",
                            "md5_2": "56271be38a80db78ef3bdfc5d9909b98"
                        },
                        [
                            "SRX9626017_SRR13191702_1.fastq.gz:md5,baaaea61cba4294ec696fdfea1610848",
                            "SRX9626017_SRR13191702_2.fastq.gz:md5,8e43ad99049fabb6526a4b846da01c32"
                        ]
                    ]
                ],
                "md5": [
                    [
                        {
                            "id": "SRX9626017_SRR13191702",
                            "single_end": false,
                            "md5_1": "89c5be920021a035084d8aeb74f32df7",
                            "md5_2": "56271be38a80db78ef3bdfc5d9909b98"
                        },
                        [
                            "SRX9626017_SRR13191702_1.fastq.gz.md5:md5,055a6916ec9ee478e453d50651f87997",
                            "SRX9626017_SRR13191702_2.fastq.gz.md5:md5,c30ac785f8d80ec563fabf604d8bf945"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,6b60ed6d5805271a1b97798e29c0635c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-28T11:51:51.301654"
    }
}